otter/log/spec.py

```diff
@@ -0,0 +1,118 @@
+"""
+Format logs with specification
+"""
+
+from copy import deepcopy
+
+from otter.constants import ServiceType
+from otter.log.formatters import IGNORE_FIELDS
+
+
+class UUID(basestring):
+    """
+    UUID
+    """
+
+
+class IPAddress(basestring):
+    """
+    IP Address
+    """
+
+
+# Allowed fields with their types
+fields = {
+    "num_servers": int,
+    "server_id": UUID,
+    "clb_id": int,
+    "ip_address": IPAddress,
+}
+
+
+# Fields that will occur in error
+error_fields = {
```

Comment:
Non-blocking comment - these work quite well for API errors.  How are non-API errors handled (e.g. tracebacks we happen to get because there might be a bug in our code, etc.)?